Initial doc for MateriaDB KV (#248)
CleverCloud/documentation: content/doc/addons/materia-db-kv.md

---
type: docs
title: MateriaDB KV
description: MateriaDB KV is a key-value database service based on our serverless offer MateriaDB
tags:
- addons
keywords:
- KV
- redis
type: docs
draft: false
---

{{< hextra/hero-subtitle >}}
  

> MateriaDB KV is a key-value database service based on our serverless offer MateriaDB.

Reviewer: ```suggestion
MateriaDB KV is a key-value database service based on Clever Cloud serverless offer MateriaDB.
```

> MateriaDB KV is a key-value database service based on our serverless offer MateriaDB.

Reviewer: ```suggestion
MateriaDB KV is a key-value database service based on Clever Cloud serverless offer MateriaDB.
```

> MateriaDB KV is built with compatibility layers allowing you to communicate with the database using well known and widely used protocols, SDKs and clients. Our first layer is made to be compatible with Redis (and it's variants as Reddict or Valkey) clients and SDKs.

Reviewer: ```suggestion
MateriaDB KV features compatibility layers allowing you to communicate with the database using well known and widely used protocols, SDKs, and clients. The first layer is compatible with Redis (and its variants as Reddict or Valkey) clients and SDKs.
```

> MateriaDB KV is built with compatibility layers allowing you to communicate with the database using well known and widely used protocols, SDKs and clients. Our first layer is made to be compatible with Redis (and it's variants as Reddict or Valkey) clients and SDKs.

Reviewer: ```suggestion
MateriaDB KV features compatibility layers allowing you to communicate with the database using well known and widely used protocols, SDKs, and clients. The first layer is compatible with Redis (and its variants as Reddict or Valkey) clients and SDKs.
```

{{< /hextra/hero-subtitle >}}

{{< callout type="warning" >}}

> **MateriaDB KV is an alpha stage product.** At Clever Cloud, we're really excited to introduce you to MateriaDB KV, currently in its alpha stage. As we fine-tune and enhance its capabilities, we invite you to join us in testing and providing valuable feedback. While MateriaDB KV holds immense potential, we advise against using it for production purposes at this time. Please be aware that during this alpha phase, all data stored within MateriaDB KV may be subject to wiping. Your insights and suggestions are crucial in shaping the future of this platform. To share your feedback, please visit us at [our MAteriaDB community page](https://github.com/CleverCloud/Community/discussions/categories/materiadb). Thank you for being a part of our journey towards innovation and improvement!

Reviewer: ```suggestion
**MateriaDB KV is an alpha stage product.** At Clever Cloud, we're really excited to introduce you to MateriaDB KV, currently in its alpha stage. As we fine-tune and enhance its capabilities, we invite you to join us in testing and providing valuable feedback. While MateriaDB KV holds immense potential, we advise against using the alpha release for production purposes. During this alpha phase, **all data stored within MateriaDB KV may be subject to wiping**. Your insights and suggestions are crucial in shaping the future of this platform. To share your feedback, please visit us at [our MAteriaDB community page](https://github.com/CleverCloud/Community/discussions/categories/materiadb). Thank you for being a part of our journey towards innovation and improvement!
```

> **MateriaDB KV is an alpha stage product.** At Clever Cloud, we're really excited to introduce you to MateriaDB KV, currently in its alpha stage. As we fine-tune and enhance its capabilities, we invite you to join us in testing and providing valuable feedback. While MateriaDB KV holds immense potential, we advise against using it for production purposes at this time. Please be aware that during this alpha phase, all data stored within MateriaDB KV may be subject to wiping. Your insights and suggestions are crucial in shaping the future of this platform. To share your feedback, please visit us at [our MAteriaDB community page](https://github.com/CleverCloud/Community/discussions/categories/materiadb). Thank you for being a part of our journey towards innovation and improvement!

Reviewer: ```suggestion
**MateriaDB KV is an alpha stage product.** At Clever Cloud, we're really excited to introduce you to MateriaDB KV, currently in its alpha stage. As we fine-tune and enhance its capabilities, we invite you to join us in testing and providing valuable feedback. While MateriaDB KV holds immense potential, we advise against using the alpha release for production purposes. During this alpha phase, **all data stored within MateriaDB KV may be subject to wiping**. Your insights and suggestions are crucial in shaping the future of this platform. To share your feedback, please visit us at [our MAteriaDB community page](https://github.com/CleverCloud/Community/discussions/categories/materiadb). Thank you for being a part of our journey towards innovation and improvement!
```

{{< /callout >}}

## Creating a MateriaDB KV database

> You can use the Clever CLI to create your MateriaDB KV addon: `clever addon create kv DATABASE_NAME`

Reviewer: ```suggestion
You can use the Clever CLI to create your MateriaDB KV add-on: `clever addon create kv DATABASE_NAME`
```

> You can use the Clever CLI to create your MateriaDB KV addon: `clever addon create kv DATABASE_NAME`

Reviewer: ```suggestion
You can use the Clever CLI to create your MateriaDB KV add-on: `clever addon create kv DATABASE_NAME`
```

> Then you can set the environement variables you need to connect to your MateriaDB KV, `source <(clever addon env addon ADDON_ID -F shell)`.

Reviewer: ```suggestion
Then you can set the environment variables you need to connect to your MateriaDB KV, `source <(clever addon env addon ADDON_ID -F shell)`.
```

> Then you can set the environement variables you need to connect to your MateriaDB KV, `source <(clever addon env addon ADDON_ID -F shell)`.

Reviewer: ```suggestion
Then you can set the environment variables you need to connect to your MateriaDB KV, `source <(clever addon env addon ADDON_ID -F shell)`.
```

Example:

```
$ clever addon create kv test

Add-on created successfully!
ID: addon_4997cfe3-f104-4d05-9fe4-xxxxxxxxx
Real ID: kv_01HV6NCSRD1TV2AJW4RKFBJ07R
Name: test

/!\ The MateriaDB KV provider is in Alpha testing phase, don't store sensitive or production grade data
You can easily use MateriaDB KV with 'redis-cli', with such commands:
source <(clever addon env addon_4997cfe3-f104-4d05-9fe4-xxxxxxxxx -F shell)
redis-cli -h $KV_HOST -p $KV_PORT
```

## Setting the environment variables

In order to connect to your MateriaDB KV addon, you need 3 parameters: the host, the port and a biscuit token. 
As explained in the response to the creation command, you can set these parameters as environment variables by doing `source <(clever addon env addon <ADDON_ID> -F shell)`. The variables set are:

- `$KV_HOST` and its alias `$REDIS_HOST`
- `$KV_PORT` and its alias `$REDIS_PORT`
- `KV_TOKEN` and its alias `$REDISCLI_AUTH`

## Using the Redis layer 

### Redis-cli usage

You can directly use the environement variables to connect to MateriaDB KV using `redis-cli`:

```bash
redis-cli -h $KV_HOST -p $KV_PORT
```

### GUI usage

> We have tested the compatibilitywith some of the most popular Redis GUI:

Reviewer: ```suggestion
We have tested the compatibility with some of the most popular Redis GUI:
```

- [Redis Insight](https://redis.com/redis-enterprise/redis-insight/)

- [Another Redis Desktop Client](https://goanother.com/)

- [Redis Commander](https://github.com/joeferner/redis-commander)

- [PX3 Redis UI](https://github.com/patrikx3/redis-ui)

- [Qredis](https://github.com/tiagocoutinho/qredis)

In all these GUI, for the authentication, leave the user empty and use as password the value of `$REDISCLI_AUTH`. 

Most other Redis GUI should also work.

### Supported types

> During this alpha stage, please note that we don't provide 100% compatibility with the Redis value types. Currently supported value types are:

Reviewer: ```suggestion
During this alpha stage, we don't provide 100% compatibility with the Redis value types. Currently supported value types are:
```

- String
- List
- Hash

### Supported commands

> During this alpha stage, please note that we don't provide 100% compatibility with the Redis API. Here you have the list of we currently supported commands:

Reviewer: ```suggestion
During this alpha stage, we don't provide 100% compatibility with the Redis API. Find below the list of we currently supported commands:
```
